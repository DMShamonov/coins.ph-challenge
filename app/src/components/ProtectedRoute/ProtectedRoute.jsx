// @flow

import * as React from 'react';
import { Route as ReactRouterDom, Redirect } from 'react-router-dom';

import ROUTES from 'constants/routes';

import type { PropsType } from './types.js.flow';

export default function ProtectedRoute({
  isLogged,
  exact = false,
  path,
  component: WrappedComponent,
}: PropsType): React.Node {
  return (
    <ReactRouterDom
      exact={exact}
      path={path}
      render={props => (
        isLogged
          ? <WrappedComponent {...props} />
          : (
            <Redirect
              to={{ pathname: ROUTES.SIGN_IN, state: { from: props.location } }}
            />
          )
      )}
    />
  );
}
