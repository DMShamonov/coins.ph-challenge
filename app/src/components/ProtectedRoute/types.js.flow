// @flow

import type { Node } from 'react';

export type PropsType = {|
  isLogged: boolean,
  exact?: boolean,
  path: string,
  component: Node,
|};
