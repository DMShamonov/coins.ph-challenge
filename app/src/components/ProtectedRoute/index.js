import verifySession from 'decorators/verifySession';

import ProtectedRoute from './ProtectedRoute';

export default verifySession(ProtectedRoute);
