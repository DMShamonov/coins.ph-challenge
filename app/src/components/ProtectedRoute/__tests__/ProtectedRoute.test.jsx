import React from 'react';
import { shallow } from 'enzyme';

import ProtectedRoute from '../ProtectedRoute';

describe('`ProtectedRoute` component should render correctly', () => {
  test('with default props', () => {
    expect(shallow((
      <ProtectedRoute
        isLogged={false}
        path="__PATH__"
        component={<p />}
      />
    ))).toMatchSnapshot();
  });

  test('when `isLogged` prop is `true`', () => {
    expect(shallow((
      <ProtectedRoute
        isLogged
        path="__PATH__"
        component={<p />}
      />
    ))).toMatchSnapshot();
  });

  test('when `exact` prop is `true`', () => {
    expect(shallow((
      <ProtectedRoute
        isLogged={false}
        exact
        path="__PATH__"
        component={<p />}
      />
    ))).toMatchSnapshot();
  });
});
