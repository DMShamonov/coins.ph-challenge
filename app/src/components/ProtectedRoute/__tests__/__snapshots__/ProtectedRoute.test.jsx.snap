// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`ProtectedRoute\` component should render correctly when \`exact\` prop is \`true\` 1`] = `
<Route
  exact={true}
  path="__PATH__"
  render={[Function]}
/>
`;

exports[`\`ProtectedRoute\` component should render correctly when \`isLogged\` prop is \`true\` 1`] = `
<Route
  exact={false}
  path="__PATH__"
  render={[Function]}
/>
`;

exports[`\`ProtectedRoute\` component should render correctly with default props 1`] = `
<Route
  exact={false}
  path="__PATH__"
  render={[Function]}
/>
`;
